packages/block-library/src/gallery/deprecated.js

```diff
@@ -19,6 +20,189 @@ import {
 } from './constants';
 
 const deprecated = [
+	{
+		attributes: {
+			images: {
+				type: 'array',
+				default: [],
+				source: 'query',
+				selector: '.blocks-gallery-item',
+				query: {
+					url: {
+						type: 'string',
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'src',
+					},
+					fullUrl: {
+						type: 'string',
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-full-url',
+					},
+					link: {
+						type: 'string',
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-link',
+					},
+					alt: {
+						type: 'string',
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'alt',
+						default: '',
+					},
+					id: {
+						type: 'string',
+						source: 'attribute',
+						selector: 'img',
+						attribute: 'data-id',
+					},
+					caption: {
+						type: 'string',
+						source: 'html',
+						selector: '.blocks-gallery-item__caption',
+					},
+				},
+			},
+			ids: {
+				type: 'array',
+				items: {
+					type: 'number',
+				},
+				default: [],
+			},
+			columns: {
+				type: 'number',
+				minimum: 1,
+				maximum: 8,
+			},
+			caption: {
+				type: 'string',
+				source: 'html',
+				selector: '.blocks-gallery-caption',
+			},
+			imageCrop: {
+				type: 'boolean',
+				default: true,
+			},
+			linkTo: {
+				type: 'string',
+			},
+			sizeSlug: {
+				type: 'string',
+				default: 'large',
+			},
+		},
+		supports: {
+			anchor: true,
+			align: true,
+		},
+		isEligible( { images } ) {
```

Comment:
Why do you need isElligible here? I believe this particular deprecation doesn't really need it?

Reply:
> > Writing a deprecation here is currently blocked by the fact that apparently, 'implicit' attributes (i.e. the ones that aren't stored as part of the block comment, but inferred through a queue) -- in this case: the images array -- aren't accessible to isEligible and migrate.
> 
> How did you arrive to this conclusion. That's not true I think or unexpected.

I put a `console.log` into both functions to output their first two arguments, and the first one only contained the attributes that were listed in the block comment.

```diff
diff --git a/packages/block-library/src/gallery/deprecated.js b/packages/block-library/src/gallery/deprecated.js
index 8542eb5bfc..b0ed9d7ea1 100644
--- a/packages/block-library/src/gallery/deprecated.js
+++ b/packages/block-library/src/gallery/deprecated.js
@@ -99,7 +99,9 @@ const deprecated = [
                        anchor: true,
                        align: true,
                },
-               isEligible( { images } ) {
+               isEligible( props, blcks ) {
+                       console.log( 'isElig', props, blcks );
+                       const { images } = props;
                        return (
                                images &&
                                images.some(
@@ -107,7 +109,8 @@ const deprecated = [
                                )
                        );
                },
-               migrate( attributes ) {
+               migrate( attributes, blcks ) {
+                       console.log( 'migrate', attributes, blcks );
                        return {
                                ...attributes,
                                images: attributes.images.map( ( image ) => {
```

![image](https://user-images.githubusercontent.com/96308/97720825-0637c900-1ac9-11eb-86b1-9b432d59855e.png)

Reply:
Because of what you said in your other [comment](https://github.com/WordPress/gutenberg/pull/26574#discussion_r515156056)? I guess if we don't need a `migrate`, we won't need an `isEligible`. But if we need a `migrate`, we'd need a way to know that the `alt` includes HTML, no?